lib/gems/pending/util/postgres_admin.rb

```diff
@@ -136,12 +136,24 @@ def self.backup_pg_compress(opts)
     opts.delete(:dbname)
 
     path = Pathname.new(opts.delete(:local_file))
-
     FileUtils.mkdir_p(path.dirname)
-    Dir.mktmpdir("vmdb_backup", path.dirname) do |dir|
-      runcmd("pg_basebackup", opts, :z => nil, :format => "t", :xlog_method => "fetch", :pgdata => dir)
-      FileUtils.mv(File.join(dir, "base.tar.gz"), path)
+
+    # Build commandline from AwesomeSpawn
+    args = {:z => nil, :format => "t", :xlog_method => "stream", :pgdata => "-"}
+    cmd  = AwesomeSpawn.build_command_line("pg_basebackup", combine_command_args(opts, args))
+
+    # Run command in a separate thread
+    read, write = IO.pipe
+    cmd_thread = Thread.new do
+      pid = Kernel.spawn(pg_env(opts), cmd, :out => write)
+      Process.wait(pid)
```

Comment:
We should probably check this return code, right?

Reply:
(I think the old logic would raise if the backup failed)

Reply:
Yeah, I might have to check what AwesomeSpawn.run! is doing for logging, and emulate that here. But I agree, we should handle here as well.